applications/GeoMechanicsApplication/custom_strategies/strategies/README.md

```diff
@@ -0,0 +1,99 @@
+# GeoMechanicsNewtonRaphsonErosionProcessStrategy class.
+
+This class implements a Newton-Raphson process for the backward erosion piping. The physics of the piping process is
+described in
+
+1. van Beek M. Backward erosion piping,initiation and progression. TUDelft PhD. Thesis, 2015.
+2. Technisch rapport Zandmeevoerende wellen. Technische Adviescommissie
+   voor de Waterkeringen, Delft, The Netherlands, 1999.
+
+## Algorithm
+
+The algorithm is implemented in the FinalizeSolutionStep function that loops over all piping elements until the erosion
+growth is finished or all piping elements are checked.
+
+### FinalizeSolutionStep function
+
+![FinalizeSolutionStep.svg](FinalizeSolutionStep.svg)
+
+This function calls other functions that are described below.
+
+### GetPipingElements function
+
+This function returns a list of all piping elements. To create the list, all elements are checked for  
+PIPE_START_ELEMENT property that holds Id of the piping starting element.
+Then elements on the list are sorted based on X-coordinate of the element begin.
+The current limitation is the following. The starting piping element has to have minimum or maximum value of the X
+coordinate. This is needed to find a direction for the piping growth.
+
+### InitialiseNumActivePipeElements function
+
+It checks the list of the elements and returns a number of elements with PIPE_ACTIVE property At the beginning the
+number is zero.
+
+### CalculateMaxPipeHeight function
+
+It calculates a maximum diameter using CalculateParticleDiameter function then multiplies it by a magic factor of 100 to
+get the height value.
+
+### CalculateParticleDiameter
+
+In case of PIPE_MODIFIED_D, the function returns
+
+$$ diameter = factor \Bigg(\frac{D_{70}}{factor}\Bigg)^{0.4}$$
+
+otherwise, $$diameter = D_{70}$$
+
+where \
+$D_{70}$ is the particle diameter of 70%-quantile of the grain size distribution by weight [m],\
+factor = $2.08\times 10^{-4}$ [m].
+
+### check_pipe_equilibrium function
+
+This function performs an iterative process towards a certain equilibrium values of the pipe heights. For each iteration
```

Comment:
Sorry to keep nagging 😄 
```suggestion
This function performs an iterative process towards certain equilibrium values of the pipe heights. For each iteration
```

Reply:
Thank you! I know that articles are very important but they do not exist in Russian language at all. ;(